Packet++/src/UdpLayer.cpp

```diff
@@ -134,6 +135,17 @@ namespace pcpp
 			m_NextLayer = SomeIpLayer::parseSomeIpLayer(udpData, udpDataLen, this, m_Packet);
 		else if ((WakeOnLanLayer::isWakeOnLanPort(portDst) && WakeOnLanLayer::isDataValid(udpData, udpDataLen)))
 			m_NextLayer = new WakeOnLanLayer(udpData, udpDataLen, this, m_Packet);
+		else if (GvcpLayer::isGvcpPort(portSrc) || GvcpLayer::isGvcpPort(portDst))
+		{
+			if (GvcpLayer::verifyRequest(udpData))
+			{
+				m_NextLayer = new GvcpRequestLayer(udpData, udpDataLen, this, m_Packet);
+			}
+			else
+			{
+				m_NextLayer = new GvcpAcknowledgeLayer(udpData, udpDataLen, this, m_Packet);
+			}
+		}
```

Comment:
It would be nicer to add a static method to `GvcpLayer` to create a GCVP layer if valid, or `PayloadLayer` otherwise. Then we can do:
```cpp
else if (GvcpLayer::isGvcpPort(portSrc) || GvcpLayer::isGvcpPort(portDst))
{
	m_NextLayer = GvcpLayer::parseGvcpLayer(udpData, udpDataLen, this, m_Packet);
}
```

Reply:
Please check the implementation in `parseGvcpLayer`.